bundles/org.openhab.binding.mielecloud/src/main/java/org/openhab/binding/mielecloud/internal/config/servlet/CreateBridgeServlet.java

```diff
@@ -0,0 +1,200 @@
+        }
+
+        String locale = getValidLocale(request.getParameter(LOCALE_PARAMETER_NAME));
+
+        logger.debug("Auto configuring Miele account using locale '{}' (requested locale was '{}')", locale,
+                request.getParameter(LOCALE_PARAMETER_NAME));
+        try {
+            Thing bridge = pairOrReconfigureBridge(locale, bridgeUid);
+            waitForBridgeToComeOnline(bridge);
+            return "/mielecloud";
+        } catch (BridgeReconfigurationFailedException e) {
+            logger.warn("{}", e.getMessage());
+            return "/mielecloud/success?" + SuccessServlet.BRIDGE_RECONFIGURATION_FAILED_PARAMETER_NAME + "=true&"
+                    + SuccessServlet.BRIDGE_UID_PARAMETER_NAME + "=" + bridgeUidString;
+        } catch (BridgeCreationFailedException e) {
+            logger.warn("Thing creation failed because there was no binding available that supports the thing.");
+            return "/mielecloud/success?" + SuccessServlet.BRIDGE_CREATION_FAILED_PARAMETER_NAME + "=true&"
+                    + SuccessServlet.BRIDGE_UID_PARAMETER_NAME + "=" + bridgeUidString;
+        }
+    }
+
+    private Thing pairOrReconfigureBridge(String locale, ThingUID bridgeUid) {
+        DiscoveryResult result = DiscoveryResultBuilder.create(bridgeUid)
+                .withRepresentationProperty(Thing.PROPERTY_MODEL_ID).withLabel(MIELE_CLOUD_BRIDGE_LABEL)
+                .withProperty(Thing.PROPERTY_MODEL_ID, MIELE_CLOUD_BRIDGE_NAME)
+                .withProperty(MieleCloudBindingConstants.CONFIG_PARAM_LOCALE, locale).build();
+        if (inbox.add(result)) {
+            return pairBridge(bridgeUid);
+        } else {
+            return reconfigureBridge(bridgeUid, locale);
+        }
+    }
+
+    private Thing pairBridge(ThingUID thingUid) {
+        Thing thing = inbox.approve(thingUid, MIELE_CLOUD_BRIDGE_LABEL, null);
+        if (thing == null) {
+            throw new BridgeCreationFailedException();
+        }
+
+        logger.debug("Successfully created bridge {}", thingUid);
+        return thing;
+    }
+
+    private Thing reconfigureBridge(ThingUID thingUid, String locale) {
+        logger.debug("Thing already exists. Modifying configuration.");
+        Thing thing = thingRegistry.get(thingUid);
+        if (thing == null) {
+            throw new BridgeReconfigurationFailedException(
+                    "Cannot modify non existing bridge: Could neither add bridge via inbox nor find existing bridge.");
+        }
+
+        ThingHandler handler = thing.getHandler();
+        if (handler == null) {
+            throw new BridgeReconfigurationFailedException("Bridge exists but has no handler.");
+        }
+
+        handler.handleConfigurationUpdate(
+                Collections.singletonMap(MieleCloudBindingConstants.CONFIG_PARAM_LOCALE, locale));
+
+        // As the parameters will not necessarily change we need to force the thing to re-initialize.
+        handler.dispose();
```

Comment:
I had a detailed look at the alternative implementation by overriding `handleConfigurationUpdate` in the `MieleBridgeHandler`. That would be copy-pasting the code from `BaseThingHandler` and removing the check whether the configuration parameters changed which isn't a better solution in imho. We really want to trigger a reinitialization here because we don't know in which state the `Thing` and ` ThingHandler` are. Isn't there a framework call we could use instead?

Reply:
My suggestion would be to put the logic that is in dispose and that you want to call here into a separate method, which you can call from here and from dispose likewise. Same for initialize. This way you can avoid calling these methods directly.

Reply:
That would be the complete initialize and dispose cycle. I decided that overwriting `handleConfiguationUpdate` is probably the better alternative and did so.